public/app/core/internationalization/index.tsx

```diff
@@ -1,26 +1,30 @@
 import i18n, { BackendModule } from 'i18next';
+import LanguageDetector from 'i18next-browser-languagedetector';
 import React from 'react';
 import { Trans as I18NextTrans, initReactI18next } from 'react-i18next'; // eslint-disable-line no-restricted-imports
 
-import { DEFAULT_LANGUAGE, LANGUAGES, VALID_LANGUAGES } from './constants';
+import { LANGUAGES, VALID_LANGUAGES } from './constants';
+
+const getLanguagePartFromCode = (code: string) => code.split('-')[0].toLowerCase();
 
 const loadTranslations: BackendModule = {
   type: 'backend',
   init() {},
   async read(language, namespace, callback) {
-    const localeDef = LANGUAGES.find((v) => v.code === language);
-
+    let localeDef = LANGUAGES.find((v) => v.code === language);
+    if (!localeDef) {
+      localeDef = LANGUAGES.find((v) => getLanguagePartFromCode(v.code) === getLanguagePartFromCode(language));
+    }
```

Comment:
In my case this was mostly for `fr` to match `fr-FR`.

I am not sure this logic can be done through [i18next configuration](https://www.i18next.com/overview/configuration-options) only since `nonExplicitSupportedLngs` doc and linked discussions seems to imply that the logic must also be implemented in BackendModule:

> `nonExplicitSupportedLngs`:  if true, will consider variants as supported when the main language is. E.g. `en-US` will be valid if `en` is in `supportedLngs`. If true and using a backend like i18next-http-backend, this can cause some request errors.

Adding this logic to the BackendModule seems the easiest way.

If in the future both `en-GB` and `en-US` should be supported, then they would be properly match by the first `find`:

```
let localeDef = LANGUAGES.find((v) => v.code === language);
```

Reply:
Yeah, but there's still some funniness going on here, I think controllable by the `load` i18next option. When we have `es-mx` it'll load both `es-mx` and also `es-es` (with a lower priority) which isn't ideal.

But, as we are today I think this is fine. 

Overall it isn't quite doing what we will _eventually_ want, but it's safe to say that's out of scope for this PR.